fetch.bs

```diff
@@ -2118,6 +2118,55 @@ clearly stipulates that <a>connections</a> are keyed on
      WebSocket saner -->
 
 
+<h3 id=http-cache-partitions>HTTP cache partitions</h3>
+
+<p>To
+<dfn lt="determine the HTTP cache partition|determining the HTTP cache partition">determine the HTTP cache partition</dfn>,
+given <var>request</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>topLevelOrigin</var> be null.
+
+ <li>
+  <p>If <var>request</var>'s <a for=request>client</a> is non-null, then:
+
+  <ol>
+   <li><p>Set <var>topLevelOrigin</var> to <var>request</var>'s <a for=request>client</a>'s
+   <a for="environment">top-level origin</a>.
+
+   <li><p>Assert: <var>topLevelOrigin</var> is an <a for=/>origin</a>.
+  </ol>
+
+ <li>
+  <p>If <var>topLevelOrigin</var> is null and <var>request</var>'s
+  <a for=request>reserved client</a> is non-null, then:
+
+  <ol>
+   <li><p>Set <var>topLevelOrigin</var> to <var>request</var>'s <a for=request>reserved client</a>'s
+   <a for="environment">top-level origin</a>.
+
+   <li>
+    <p>If <var>topLevelOrigin</var> is null, then set <var>topLevelOrigin</var> to
+    <var>request</var>'s <a for=request>reserved client</a>'s
+    <a for="environment">top-level creation URL</a>'s <a for=url>origin</a>.
+
+    <p class=note>This happens for top-level navigations.
+  </ol>
+
```

Comment:
How does https://github.com/whatwg/fetch/pull/943#discussion_r410351876 work with this? As per the first condition it will be client's topLevelOrigin, but wouldn't that map to A/B instead of C?

Reply:
Thanks. We should use the reserved client if non-null and only then use the regular client if it's still null. I'll fix that.